processing/src/main/java/io/druid/segment/data/GenericIndexed.java

```diff
@@ -407,35 +216,56 @@ public T get(int index)
   @Override
   public int indexOf(T value)
   {
-    return bufferIndexed.indexOf(value);
+    return indexOf(this, value);
+  }
+
+  private int indexOf(Indexed<T> indexed, T value)
```

Comment:
I'm not sure if this actually matters for performance, but you've technically now pushed indirected calls into the tight loop of `indexOf` and will cause the JVM to have to disambiguate on every single call to `indexed`.  The previous implementation had a single indirection to figure out the type of `bufferIndexed` and then all of the code was local to the implementation.  This previous implementation is generally better for playing with JIT.